### ISSUE TITLE
Provide a concrete example how to use vault-anyconfig

### ISSUE THREAD

**Issue by czchen** [tomtom-international/vault-anyconfig#13]
* vault-anyconfig version: 0.2.0
* Python version: 3.7
* Operating System: Debian 9

### Description

Please help to provide a concrete example for how to use `vault-anyconfig`. Currently we use `anyconfig.load` to load configuration, but we cannot read any credential from vault.

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```

**Comment by czchen** [comment]
Hi @eugene-davis ,

Thanks for the hint. However, since we use other feature like reading multiple config files in anyconfig. Is there any way to use original `anyconfig.load` with `vault-anyconfig` feature? A simple example would be appreciated.

**Comment by czchen** [comment]
Hi @eugene-davis,

Thanks for the advise. However, since we use other features like loading multiple config files in original anyconfig. Is it possible to use original `anyconfig.load` with `vault-anyconfig` feature? If that is possible, a simple example would be appreciated.

**Comment by eugene-davis** [comment]
@czchen `vault-anyconfig` basically is a thin wrapper around `anyconfig`, so you should be able to pass the same file spec you would use in `anyconfig.load` to `vault-anyconfig.load`.

For a simple example:
```python
from vault_anyconfig.vault_anyconfig import VaultAnyConfig

# Configure as appropriate to connect to Vault. You can of course also put this in a configuration file (readable by anyconfig)
config_client = VaultAnyConfig(url="vault-url")

# You can also directly use the HVAC login if preferable, this is just a convenience function for loading the vault credentials directly from a file (readable by anyconfig)
config_client.auth_from_file("vault_creds.yaml")

# This call will pass on any arguments given to it (except process_secret_files) to `anyconfig.load`
config = config_client.load("*.yaml")
```

You can see in https://github.com/tomtom-international/vault-anyconfig/blob/master/vault_anyconfig/vault_anyconfig.py#L133 that `vault-anyconfig.load` directly calls `anyconfig.load`, and simply processes the result.

Basically, you need to initialize a Vault connection with two steps, then you would replace your `anyconfig` calls with `vault-anyconfig` calls.

**Comment by czchen** [comment]
Hi @eugene-davis 

We have `hvac.exceptions.InvalidRequest: missing client token` when loading config via `VaultAnyConfig`. Any idea why?

The following is source code to load config

```python
    config_client = VaultAnyConfig(vault_config_file=f'{path}/config/vault.yml')

    config = config_client.load([
        f'{path}/default_config.yml',
    ], ac_ignore_missing=True)
```

The following is content of `vault.yml`

```
vault_config:
  url: https://example.com

vault_creds:
  role_id: <role_id>
  secret_id: <secret_id>
  auth_method: approle

vault_secrets:
  app.config.SQLALCHEMY_DATABASE_URI: <path>
```

**Comment by eugene-davis** [comment]
You didn't authenticate the Vault client (this is done separately from the construction to keep compatibility with the base client).
`config_client.auth_from_file("vault.yaml")` needs to be called before you use `load`.
